### PR TITLE
Shameera/add scalafmt to scalafix

### PR DESCRIPTION
**Pull request by syodage**
Fixes: https://github.com/spotify/scio/issues/2490

- [x] Add scalafmt support to scalafix subproject
- [x] Use the same `.scalafmt.conf` file in the main project
- [x] Add check to the circleci's check job.

**Note** this PR is based on #2468

### CODE REVIEW

**Comment by syodage** [comment]
@regadas , you are right there were some failures with scalafix, but managed to fix those with little tweaking. In future if we don't need scalafmt to format our testing rules we can just disable the formatting with `// format: off` and enbale it back `// format: on`

**Comment by regadas** [comment]
Yup i know that it's possible, but i still think it's better to just exclude them... the same way we are tweaking it around the imports it will probably happen for other cases and we end up with a bunch of on/off's

File: .scalafmt.conf [hunk 1]
```diff
@@ -25,5 +25,7 @@ rewrite {
 
 project.excludeFilters = [
   "MultiJoin.scala",
-  "TupleCoders.scala"
+  "TupleCoders.scala",
+  "scalafix/input/*",
+  "scalafix/output/*"
```
Comment (syodage) [inline .scalafmt.conf]: These two filter out followings. 
`scalafix/input`
`scalafix/input-0_8`
`scalafix/output`
`scalafix/output-0_8`

**Comment by syodage** [comment]
@regadas updated to run scalafmt only on `rules`

**Comment by regadas** [comment]
@syodage CI still complains about the scalafix output

**Pull request merged**